### DISCUSSION TITLE
Anycubic i3 mega ESP8266 not working

### DISCUSSION THREAD

**Discussion by faheemaziz41** [luc-github/ESP3D#810] [Questions]
I am having issues with my i3 mega and the 8266 as when I refresh the printer configuration nothing comes up. I have tried to change the serial_port and baud rate although its still not working. I followed this tutorial https://github.com/knutwurst/Marlin-2-0-x-Anycubic-i3-MEGA-S/wiki/WiFi-Installation-%28english%29 and wiring but no luck. I have attached the files to review! Any help would be amazing. These is the ESP I have: https://www.az-delivery.uk/en/products/nodemcu
[Marlin-AI3M-BLTouch-master.zip](https://github.com/luc-github/ESP3D/files/9492248/Marlin-AI3M-BLTouch-master.zip)

**Comment by luc-github** [comment]
first your values are not following the tutorial :
```
#define SERIAL_PORT 0
#define SERIAL_PORT_2 1

/**
 * This setting determines the communication speed of the printer.
 *
 * 250000 works in most cases, but you might try a lower speed if
 * you commonly experience drop-outs during host printing.
 * You may try up to 1000000 to speed up SD file transfer.
 *
 * :[2400, 9600, 19200, 38400, 57600, 115200, 250000, 500000, 1000000]
 */
#define BAUDRATE 25000
#define BAUDRATE_2 9600
```
`BAUDRATE` should be `250000` not `25000` which is not supported
according tutorial `SERIAL_PORT_2` should be `2` not `1`

also instead of `BAUDRATE_2` `9600` please use `115200` which default value of ESP3D in case of you did not changed it in webUI

Give a try to changes I suggest

**Comment by faheemaziz41** [comment]
Thanks for the advice, still no luck with that setup either :(. Heres the updated marlin files: 
[Marlin-AI3M-BLTouch-master.zip](https://github.com/luc-github/ESP3D/files/9492629/Marlin-AI3M-BLTouch-master.zip)
<img width="1890" alt="Screenshot 2022-09-06 024311" src="https://user-images.githubusercontent.com/61152337/188529379-58c2c46d-7cb0-4dda-9cff-1016d2f0779c.png">

**Reply by luc-github** [reply]
do others commands works ? like M105

**Reply by luc-github** [reply]
also do a [ESP420] in terminal and share the output

**Reply by faheemaziz41** [reply]
M105 didn't return the temp reading. Not sure if that's what I was meant to do for the ESP420 command but this is what I got when I inputted in into my 3D printer code terminal.
<img width="756" alt="esp420" src="https://user-images.githubusercontent.com/61152337/188624456-d92e8445-07b2-46ff-bdfe-157a72f50d3c.png">
<img width="481" alt="M105" src="https://user-images.githubusercontent.com/61152337/188624459-14450504-e749-4845-b4fa-c2820bb83e26.png">

**Reply by luc-github** [reply]
type [ESP420] in webui terminal
with the brackets

**Reply by faheemaziz41** [reply]
Here is the result:
[ESP420]
{"chip_id":"14157538","cpu":"160","freemem":"36.31 KB","SDK":"2.2.1(cfd48f3)","flash_size":"4.00 MB","update_size":"534.87 KB","spiffs_size":"1.86 MB","baud_rate":"115200","sleep_mode":"Modem","channel":"11","phy_mode":"11g","web_port":"80","data_port":"8888","hostname":"FAZESP","active_mode":"STA (AC:0B:FB:D8:06:E2)","connected_ssid":"VM7342653","connected_signal":"50%","ip_mode":"DHCP","ip":"192.168.0.38","gw":"192.168.0.1","msk":"255.255.255.0","dns":"194.168.4.100","disabled_mode":"AP (AE:0B:FB:D8:06:E2)","captive_portal":"Enabled","ssdp":"Enabled","netbios":"Disabled","mdns":"Enabled","web_update":"Enabled","pin recovery":"Disabled","autentication":"Disabled","target_fw":"Marlin","M117_output":"Enabled","Notifications":"Disabled","Serial_output":"Enabled","Websocket_output":"Enabled","TCP_output":"Enabled","fw":"2.1.1.b10 ESP8266/8586"}

**Reply by luc-github** [reply]
ok looks ok, do you have picture of your esp wired to your board ?

**Reply by faheemaziz41** [reply]
Ive used a buck converter to step down a 5v and GND signal to 3.3v to power the esp. Then wired the TX and RX as shown in the tutorial image. Ive also attached the pin diagram for the board which is wired to EXP1.
![Buck converter](https://user-images.githubusercontent.com/61152337/188629735-96345006-1f75-4782-9a8d-04af549502e6.png)
![ESP](https://user-images.githubusercontent.com/61152337/188629768-8a066817-e912-4395-9430-6a6caf51e995.png)
![ESP2](https://user-images.githubusercontent.com/61152337/188629813-b77ae4ce-1764-4a99-8f20-ca0c6d605be3.jpg)
![ESP_02](https://user-images.githubusercontent.com/61152337/188629845-5230ad47-bc3c-40d8-83ab-5db599831ee1.jpg)
![XjIZe](https://user-images.githubusercontent.com/61152337/188629862-f0f03bd7-d803-41e1-a6b1-4db7cb6ee7ce.jpg)

**Reply by luc-github** [reply]
first test exchange tx/rx on esp, may be you tested that already?

**Reply by faheemaziz41** [reply]
Yep tried that before and just tried again, no luck :(

**Reply by luc-github** [reply]
2nd test 
connect GND  by passing your buck converter:  esp GND->printer GND

**Reply by faheemaziz41** [reply]
No still no response. How would I know that the EXP1 D17 and D16 are the correct TX and RX pins?

**Reply by luc-github** [reply]
I do not know this board I cannot answer - but I see there is an UART 3 header - why not using it ?

**Reply by luc-github** [reply]
here pinout of MEGA https://community.element14.com/products/arduino/w/documents/2969/arduino-mega-2560-rev3-pinout-atmega2560-pin-mapping-eagle-files-schematics-and-more
seems correct

**Reply by faheemaziz41** [reply]
Just wired it up and changed the serial port to 3 (UART3) in configuration.h and still no response

**Reply by luc-github** [reply]
<s>Sorry I am dry 
I suggest you to contact https://github.com/knutwurst/Marlin-2-0-x-Anycubic-i3-MEGA-S because I am sure it is related to some setting in his FW</s>
See next response

**Reply by luc-github** [reply]
Ok I found the root cause
you use a FW which is not https://github.com/knutwurst/Marlin-2-0-x-Anycubic-i3-MEGA-S
and does not support SERIAL_PORT_2 ...
So it is normal nothing works ...

**Reply by luc-github** [reply]
Firmware, in your case the Marlin Firmware sources, the sources you shared are not the same as the tutorial, and they do not support more than one serial port

**Reply by faheemaziz41** [reply]
So I should use the FW version he is using and it will work?

**Reply by luc-github** [reply]
yes or original Marlin Firmware should also work I think

**Reply by luc-github** [reply]
at least now you know why it was not working :D

**Reply by faheemaziz41** [reply]
I've just setup the code from knutwurst and now the ESP is making a beep when I boot the printer. Although it still does not seem to connect. I've also just tried changing the first serial port and Buad rate to the ESP however it still doesn't work :(

**Comment by luc-github** [comment]
a beep ? how come ? nodemcu does not have any buzzer/speaker

For the communication issue check all steps I mentionned earlier, also use same baud rate as defined in tutorial to be sure it is same as described and do not forget to update ESP3D baud rate and restart the board

**Reply by faheemaziz41** [reply]
I guess the 3D printer is making the beep because of the firmware change. I've tried 9600, 115200, 250000 and 500000 baud rate with this marlin setup with no luck yet

**Reply by luc-github** [reply]
when you change baud rate in esp3d , restart the esp board
and no need to try all it won't change anything

**Reply by faheemaziz41** [reply]
These are the modules I bought, should they work fine with this application? 
https://www.amazon.co.uk/AZDelivery-NodeMCU-Module-ESP8266-Parent/dp/B07ZRW28L4
Also reset ESP after setup changes and still not working

**Reply by faheemaziz41** [reply]
What versions of Marlin will work with this?

**Reply by luc-github** [reply]
2.x is supporting up to 3 serial ports now

**Reply by faheemaziz41** [reply]
Just tried with 2.0-x and seems to still be unresponsive :(
